src/consumption.cpp

```diff
@@ -732,6 +732,19 @@ void player::consume_effects( item &food, bool rotten )
             fun = 1;
         }
     }
+    //Food is made out of meat (use ALLERGEN_MEAT, not CARNIVORE_OK (the latter has eggs and milk),
+    //Reduces bad morale penalties on food only made of meat.
+    if( has_trait( "CARNIVORE" ) && food.has_flag( "ALLERGEN_MEAT" ) &&
```

Comment:
CARNIVORE_OK includes eggs and milk, while they are okey for the carnivore to eat, I  didn't think they warranted the morale boost.